src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,48 @@
 package core.mate.academy.service;
 
+import core.mate.academy.impl.BulldozerProducer;
+import core.mate.academy.impl.ExcavatorProducer;
+import core.mate.academy.impl.TruckProducer;
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class type) {
+        MachineProducer<T> producer;
+        if (type == Bulldozer.class) {
+            producer = (MachineProducer<T>) new BulldozerProducer();
+            return producer.get();
```

Comment:
Casting to `MachineProducer<T>` is unsafe here. Instead, `MachineProducer` interface should be parameterized to prevent this possibility.